google/cloud/spanner/client.h

```diff
@@ -128,10 +130,31 @@ class Client {
    * with unit testing, callers may create fake/mock `Connection` objects that
    * are injected into the `Client`.
    */
-  explicit Client(std::shared_ptr<Connection> conn, ClientOptions opts = {})
+  explicit Client(std::shared_ptr<Connection> conn, Options opts = {})
       : conn_(std::move(conn)), opts_(std::move(opts)) {}
 
-  /// No default construction. Use `Client(std::shared_ptr<Connection>)`
+  /**
+   * Constructs a `Client` object using the specified @p conn and @p opts.
+   *
+   * Note that the previous constructor, which uses the new way to represent
+   * options of all varieties (`google::cloud::Options`), is now preferred.
+   */
+  explicit Client(std::shared_ptr<Connection> conn, ClientOptions const& opts)
+      : conn_(std::move(conn)), opts_(opts) {}
+
+  /**
+   * Constructs a `Client` object using the specified @p conn and an empty
+   * initializer list.
+   *
+   * @note This constructor exists solely for backwards compatibility. It
+   *     prevents existing code that uses `Client(conn, {})` from breaking
+   *     due to ambiguity introduced by the `Options` overload.
+   */
```

Comment:
In pubsub, I introduced this constructor as `@deprecated`, to scare away any users reading the documentation. In the long run, we do want to remove the function, but, we also want `Client(conn, {})` to be valid code that calls the `Options` constructor. (So I'm not sure using `@deprecated` was correct).

Changing from a `@note` to a `@warning` is also possible. I'll let you decide if you think either of these are improvements.

Reply:
Yeah, I don't think `@deprecated` is best for the reasons you cite.  And indeed, `@warning` falls to the same logic methinks.  The `@note` is really just an answer to "What's this?", and it all goes away when `ClientOptions` does.